### PR TITLE
Update policy mapping design document

### PR DESCRIPTION
**Pull request by larkinscarvalho**

### CODE REVIEW

File: designs/manager_policy_mapping/policy_mapping.md [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-This document describes the proposed changes for policy mapping workflow with IPs/MACs.
+This document describes the proposed changes for multi-policy mapping workflow.
 
 ## Background
 
 Currently, we have two ways to determine which policy gets applied to which workload:
 
 - Provide the VNF security manager with encapsulation tag
-- Provide the VNF security manager with IP/MAC
+- Provide the VNF security manager with list of policies
```
Comment (arvindn05) [inline designs/manager_policy_mapping/policy_mapping.md]: IP/Mac is still relavent. 

Reword to -Provide the VNF security manager with IP/Mac and a list of polices/tags to associate with them

File: designs/manager_policy_mapping/policy_mapping.md [hunk 2]
```diff
@@ -26,7 +26,7 @@ The appliance security manager will not delete a security group with one or more
     "virtualSystemId": 0,
     "name": "string",
     "policyId": 0,
-    "managerSecurityGroupId": "string", // This new property is being added to store the security group id returned by the manager
+    "managerSecurityGroupId": "string", // This new property is added to persist the manager security group id
     "failurePolicyType": "FAIL_OPEN",
     "order": 0,
     "policies": [
```
Comment (arvindn05) [inline designs/manager_policy_mapping/policy_mapping.md]: add comment on how this is now a list.

Also add information on what API is impacted with this change(with path and whether its a PUT/GET/POST etc

File: designs/manager_policy_mapping/policy_mapping.md [hunk 3]
```diff
@@ -92,20 +105,59 @@ public interface ManagerSecurityGroupInterfaceApi {
...
+     */
+    String getId();
+
+    /**
+     * @return the name of the policy defined in the security manager
+     */
+    String getName();
+
+    /**
+     * @return the identifier of the domain, the policy belongs to in the security manager
```
Comment (emanoelxavier) [inline designs/manager_policy_mapping/policy_mapping.md]: no `,` needed

File: designs/manager_policy_mapping/policy_mapping.md [hunk 4]
```diff
@@ -92,20 +105,59 @@ public interface ManagerSecurityGroupInterfaceApi {
...
 #### SDN Controller SDK
 No changes to SDN Controller SDK are required.
 
 ### VNF Security Manager Plugins
-Updating the plugins implementing these APIs is out of scope for this feature, with the exception of the security-mgr-sample-plugin which will be modified accordingly.
+Updating the plugins implementing these APIs is out of scope for this feature, with the exception of the security-mgr-sample-plugin which will be modified to support multiple policy mapping.
 
 ### OSC UI
-No changes to the OSC UI are required.
+Multi-policy mapping is out of scope.
```
Comment (emanoelxavier) [inline designs/manager_policy_mapping/policy_mapping.md]: scope. -> scope for the upcoming release.

**Comment by emanoelxavier** [comment]
@larkinscarvalho also please include the pic arvind shared above with some explanation of the components or flow. It will be highly helpful.

**Comment by arvindn05** [comment]
@larkinscarvalho please update the pull request for the comments.

File: designs/manager_policy_mapping/policy_mapping.md [hunk 5]
```diff
@@ -1,34 +1,50 @@
...
+- Providing the VNF security manager with the IP/Mac information and a list of polices/tags to be associated
 
 If the manager is supporting [Policy Mapping](https://github.com/opensecuritycontroller/security-mgr-api/blob/master/src/main/java/org/osc/sdk/manager/Constants.java#L70) and [Security Group](https://github.com/opensecuritycontroller/security-mgr-api/blob/master/src/main/java/org/osc/sdk/manager/Constants.java#L60), we provide the manager with the security group name and members (their IPs and MACs) information to bind the policy.
 
+**Policy mapping workflow**
+
+![](./images/policy_mapping.png)
+
+OSC connects to the manager by adding a manager connector and syncs the domains and the policies avaiable with the manager. The security group on OSC contains workload information. As a part of the binding process, OSC provides the security manager with workload IP/MAC information and the list of policies or tags to be associated with the workload. The security manager provides the supplied information to the security appliance. The security appliance applies the right policies to the workload based on the IP/MAC information.
```
Comment (arvindn05) [inline designs/manager_policy_mapping/policy_mapping.md]: split this up into bullet or labelled points based on the graphic above.

Also in the diagram, replace "binding" from step 3 with "associate list of policies/tags with Security group workloads"

**Comment by larkinscarvalho** [comment]
@sudhirappaji review comments are not seen on file. 

Currently in the plugin SG and SGI is one-to-one mapping. or OSC and plugin implementation need to differ? not clear.
**Yes. The OSC and plugin implementation differ.**

Can you provide the pointer to encapsulation tag? it is not clear what this tag contain?
**Policy and the tag is the same. In PAN manager the tag is the policy.
However, the encapsulation tag is different from the tag information we are referring to.** 

What are the additional details?
**By additional details I meant the normal response we receive with the GET request. 
I have only highlighted the change i.e. now we will receive the policyIds as list**

**Comment by larkinscarvalho** [comment]
@arvindn05 ready to merge.
Thanks

**Pull request merged**